File: duplicate-zeros/duplicate-zeros.ts
/**
 Do not return anything, modify arr in-place instead.
 */
function duplicateZeros(arr: number[]): void {
  for(let i = arr.length - 1; i >= 0; i--) {
    if(arr[i] === 0) {
      for(let j = arr.length - 1; j > i; j--) {
        arr[j] = arr[j - 1]
      }
    }
  }
 
};

File: duplicate-zeros/duplicate-zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        i = 0 
        while i < len(arr):
          if arr[i] == 0:
            j = len(arr) - 1
            while j > i:
              arr[j] = arr[j - 1]
              j-=1
            i+=1
          i+=1
        """
        Do not return anything, modify arr in-place instead.
        """
        